src/test/java/test/methodinterceptors/issue1726/PriorityInterceptor.java

```diff
@@ -0,0 +1,48 @@
+package test.methodinterceptors.issue1726;
+
+import org.testng.IMethodInstance;
+import org.testng.IMethodInterceptor;
+import org.testng.ITestContext;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class PriorityInterceptor implements IMethodInterceptor {
+    public static List<String> ordered = new ArrayList<>();
+
+    public List<IMethodInstance> intercept(List<IMethodInstance> methods,
+                                           ITestContext context) {
+        Comparator<IMethodInstance> comparator = new Comparator<IMethodInstance>() {
+
+            private int getPriority(IMethodInstance mi) {
+                int result = 0;
+                Method method = mi.getMethod().getConstructorOrMethod().getMethod();
+                Priority a1 = method.getAnnotation(Priority.class);
+                if (a1 != null) {
+                    result = a1.value();
+                } else {
+                    Class<?> cls = method.getDeclaringClass();
+                    Priority classPriority = cls.getAnnotation(Priority.class);
+                    if (classPriority != null) {
+                        result = classPriority.value();
+                    }
+                }
+                return result;
+            }
+
+            public int compare(IMethodInstance m1, IMethodInstance m2) {
+                return getPriority(m1) - getPriority(m2);
+            }
+
+        };
+        methods.sort(comparator);
+        methods.forEach(m -> {
```

Comment:
If you want to use `Stream`, you should use `map()`,  `sort()` and `collect()` too.

Reply:
This wasn't intentional. When I created a sample that can be used to reproduce the issue, I ended up using Stream. I am for now focussing on the JDK8 Stream features to be used here. But let me ensure that I use all of them or get rid of Stream here.

Reply:
@juherr - I have amended the commit and pushed in changes to use all JDK8 features in this class.